flink-runtime/src/main/java/org/apache/flink/runtime/messages/ThreadInfoSample.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.messages;
+
+import java.io.Serializable;
+import java.lang.management.LockInfo;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+
+/**
+ * A serializable wrapper around {@link java.lang.management.ThreadInfo} that excludes {@link
+ * java.lang.management.LockInfo}-based non-serializable fields.
+ */
+public class ThreadInfoSample implements Serializable {
+
+    private final String threadName;
+    private final long threadId;
+    private final long blockedTime;
+    private final long blockedCount;
+    private final long waitedTime;
+    private final long waitedCount;
+    private final String lockName;
+    private final long lockOwnerId;
+    private final String lockOwnerName;
+    private final boolean inNative;
+    private final boolean suspended;
+    private final Thread.State threadState;
+    private final StackTraceElement[] stackTrace;
```

Comment:
Do we really need all these things?

Reply:
Not for the Flame Graphs, but I was thinking that they maybe could be helpful if we further extend our debugging toolkit. Remove unused for now?

Reply:
Done.